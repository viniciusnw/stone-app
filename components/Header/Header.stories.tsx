import React from "react";
import { Header } from "@Components";

export const Default = {
  args: {
    type: "default",
  },
};

export const NoBack = {
  args: {
    type: "no_back",
  },
};

export const Cart = {
  args: {
    type: "cart",
  },
};

export default {
  title: "Header",
  argTypes: {
    type: {
      options: ["default", "no_back", "black", "cart"],
      control: { type: 'radio' },
    },
  },
  component: ({ type }: any) => {
    switch (type) {
      case "default":
        return <Header back={() => true} menu={() => true} />;
      case "no_back":
        return <Header back={false} menu={() => true} />;
      case "cart":
        return  <Header back={() => true} cart title="Checkout" />;
      default:
        return <Header back={() => true} menu={() => true} />;
    }
  },
};
