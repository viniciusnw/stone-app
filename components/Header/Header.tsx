import React from "react";
import styled from "styled-components/native";
import { View, Text, TouchableOpacity } from "react-native";
import Icon from "@Components/Icon";
import Logo from "@Components/Logo";
import Label from "@Components/Label";

const StyledContainer = styled(View)`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 62px;
  border: 1px solid transparent;
  border-bottom-color: ${({ theme }) => theme.color.border};
  padding: 0 ${({ theme }) => theme.spacing.regular};
`;

const StyledTitle = styled(Text)`
  color: ${({ theme }) => theme.color.black};
  font-weight: 700;
  font-size: 20px;
  max-width: 200;
  text-align: center;
`;

const StyledFake = styled(View)`
  width: 30px;
`;

const CartQty = styled(View)`
  position: absolute;
  top: -15;
  right: -10;
`;

const StyledBack = styled(TouchableOpacity)``;

const StyledCart = styled(TouchableOpacity)``;

const Cart = ({ cart, cartQty }: any) => {
  return (
    <StyledCart onPress={cart}>
      <Icon icon="cart" size={26} />

      {!!cartQty && (
        <CartQty>
          <Label rounded label={cartQty} color="danger" />
        </CartQty>
      )}
    </StyledCart>
  );
};

const Header = (props: any) => {
  const { back, cart, title, cartQty } = props;

  return (
    <StyledContainer>
      {back ? (
        <StyledBack onPress={back}>
          <Icon icon="left" size={22} />
        </StyledBack>
      ) : (
        <StyledFake />
      )}

      <StyledTitle numberOfLines={2}>{title ? title : <Logo />}</StyledTitle>

      {cart ? <Cart cart={cart} cartQty={cartQty} /> : <StyledFake />}
    </StyledContainer>
  );
};

export default Header;
