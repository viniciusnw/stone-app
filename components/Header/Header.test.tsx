import React from "react";
import renderer from "react-test-renderer";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import Header from "./Header";

jest.useFakeTimers();

test("Header", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <Header back={() => true} menu={() => true} />
        <Header back={false} menu={() => true} />
        <Header back={() => true} cart title="Checkout" />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
