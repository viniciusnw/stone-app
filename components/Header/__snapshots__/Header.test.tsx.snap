// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#ddd",
        "borderColor": "transparent",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 62,
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 0,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#111111",
          "fontSize": 20,
          "fontWeight": "700",
          "maxWidth": 200,
          "textAlign": "center",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "FredokaOne-Regular",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(0,168,104)",
                "fontSize": 32,
                "fontWeight": "700",
              }
            }
          >
            Stone
          </Text>
          <Text
            style={
              {
                "color": "rgb(10,245,61)",
                "fontSize": 32,
                "fontWeight": "700",
              }
            }
          >
            Ton :)
          </Text>
        </Text>
      </View>
    </Text>
    <View
      style={
        {
          "width": 30,
        }
      }
    />
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#ddd",
        "borderColor": "transparent",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 62,
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 0,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "width": 30,
        }
      }
    />
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#111111",
          "fontSize": 20,
          "fontWeight": "700",
          "maxWidth": 200,
          "textAlign": "center",
        }
      }
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "FredokaOne-Regular",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgb(0,168,104)",
                "fontSize": 32,
                "fontWeight": "700",
              }
            }
          >
            Stone
          </Text>
          <Text
            style={
              {
                "color": "rgb(10,245,61)",
                "fontSize": 32,
                "fontWeight": "700",
              }
            }
          >
            Ton :)
          </Text>
        </Text>
      </View>
    </Text>
    <View
      style={
        {
          "width": 30,
        }
      }
    />
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#ddd",
        "borderColor": "transparent",
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 62,
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 0,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#111111",
          "fontSize": 20,
          "fontWeight": "700",
          "maxWidth": 200,
          "textAlign": "center",
        }
      }
    >
      Checkout
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text />
    </View>
  </View>,
]
`;
